packages/flutter/lib/src/cupertino/date_picker.dart

```diff
@@ -948,9 +948,15 @@ class _CupertinoDatePickerDateTimeState extends State<CupertinoDatePicker> {
       _getEstimatedColumnWidth(_PickerColumnType.hour),
       _getEstimatedColumnWidth(_PickerColumnType.minute),
     ];
+
+    // Swap the hours and minutes if RTL to ensure they are in the correct position.
+    final bool isRtl = Directionality.of(context) == TextDirection.rtl;
```

Comment:
There's nothing wrong with this but personally, I'd write:
```dart
final List<_ColumnBuilder> pickerBuilders = Directionality.of(context) == TextDirection.rtl
  ? <_ColumnBuilder>[_buildMinutePicker, _buildHourPicker]
  : <_ColumnBuilder>[_buildHourPicker, _buildMinutePicker];

```

Reply:
Yeah I agree, this is cleaner